services/src/main/java/org/keycloak/email/DefaultEmailSenderProvider.java

```diff
@@ -185,16 +188,29 @@ protected String retrieveEmailAddress(UserModel user) {
     }
 
     private void setupTruststore(Properties props) {
-        JSSETruststoreConfigurator configurator = new JSSETruststoreConfigurator(session);
-
-        SSLSocketFactory factory = configurator.getSSLSocketFactory();
-        if (factory != null) {
-            props.put("mail.smtp.ssl.socketFactory", factory);
-            if (configurator.getProvider().getPolicy() == HostnameVerificationPolicy.ANY) {
-                props.setProperty("mail.smtp.ssl.trust", "*");
-                props.put("mail.smtp.ssl.checkserveridentity", Boolean.FALSE.toString());
-            }
-        }
+         JSSETruststoreConfigurator configurator = new JSSETruststoreConfigurator(session);
+         if (configurator.getProvider().getPolicy() == HostnameVerificationPolicy.ANY) {
+             props.setProperty("mail.smtp.ssl.trust", "*");
+             props.put("mail.smtp.ssl.checkserveridentity", Boolean.FALSE.toString());
+             TrustManager[] trustAllCerts = new TrustManager[]{
+                 new X509TrustManager() {
+                     public X509Certificate[] getAcceptedIssuers() {
+                         return null;
+                     }
+                     public void checkClientTrusted(X509Certificate[] certs, String authType) {
+                     }
+                     public void checkServerTrusted(X509Certificate[] certs, String authType) {
+                     }
+                 }
+             };
+             props.put("mail.smtp.ssl.socketFactory", trustAllCerts);
```

Comment:
This doesn't look right - do you mean to set the TrustManager array to the socketFactory property?

Reply:
My bad i didn't know i didn't need an array to set socketFactory property. 
fixed here : https://github.com/keycloak/keycloak/blob/3379ed347702fc390dfdd53b8376b36bfe96a831/services/src/main/java/org/keycloak/truststore/JSSETruststoreConfigurator.java#L55